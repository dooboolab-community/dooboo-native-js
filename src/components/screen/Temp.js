// @flow
import React, { Component } from 'react';
import {
  Platform,
  StatusBar,
  StyleSheet,
  TouchableHighlight,
  TouchableOpacity,
  Image,
  ScrollView,
  Text,
  View,
  FlatList,
  InteractionManager,
} from 'react-native';

import { ratio } from '../../utils/Styles';
import {
  IC_MASK,
} from '../../utils/Icons';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

type Props = {
}

type State = {
}

class Page extends Component<Props, State> {
  static navigationOptions = {
    headerTitle: <Text style={{
      fontSize: 18,
    }}>Temp</Text>,
  };

  render() {
    return (
      <View style={styles.container}>
        <Text>Temporary Page</Text>
      </View>
    );
  }
}

export default Page;
