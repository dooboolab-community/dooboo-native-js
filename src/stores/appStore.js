import { observable } from 'mobx';
import { NavigationActions } from 'react-navigation';

import User from '../models/User';

class ObservableListStore {
  @observable rootNavigatorActionHorizontal: boolean = true;
  @observable user: User = new User();
}

const observableListStore = new ObservableListStore();
export default observableListStore;
