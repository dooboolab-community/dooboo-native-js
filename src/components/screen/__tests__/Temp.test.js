import 'react-native';
import * as React from 'react';
import Temp from '../Temp';
import appStore from '../../../stores/appStore';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';

describe('Temp', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<Temp store={appStore}/>).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('component test', () => {
    const wrapper = shallow(
      <Temp store={appStore} />,
    );

    it('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
