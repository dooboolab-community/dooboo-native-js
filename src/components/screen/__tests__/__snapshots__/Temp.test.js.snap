// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temp component test renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text>
    Temporary Page
  </Text>
</View>
`;

exports[`Temp renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text>
    Temporary Page
  </Text>
</View>
`;
